kernel/work.c

```diff
@@ -853,6 +921,18 @@ int k_work_queue_stop(struct k_work_q *queue, k_timeout_t timeout)
 	return 0;
 }
 
+#if defined(CONFIG_WORKQUEUE_WORK_TIMEOUT)
+bool k_work_queue_is_blocked(struct k_work_q *queue)
+{
+	return flag_test(&queue->flags, K_WORK_QUEUE_BLOCKED_BIT);
+}
+
+bool k_sys_work_queue_is_blocked(void)
+{
+	return flag_test(&k_sys_work_q.flags, K_WORK_QUEUE_BLOCKED_BIT);
+}
```

Comment:
Not really seeing the point of this as an application API?  I mean, you're not supposed to be blocked at all.  One doesn't normally write `is_my_code_broken()` predicates, nor call them in working code.

Reply:
This is for use in tests, can move it to an internal header :)

Reply:
If we use k_oops() on block this API can be removed entirely

Reply:
API removed, thread is aborted, so a user who cares could join or check if the work queue thread is running :)